Add docs for SHOW EXPERIMENTAL_RANGES

rmloveland opened the pull request:
Fixes #2282 

Related:
- https://github.com/cockroachdb/cockroach/pull/14390
- https://github.com/cockroachdb/cockroach/pull/24696

Specific changes:

- Add a statement page for `SHOW EXPERIMENTAL_RANGES`

- `s/TESTING_RANGES/EXPERIMENTAL_RANGES/g` wherever it was previously
  mentioned, such as in `SPLIT AT`, partitioning docs, and some of the
  trainings

- Add `SHOW EXPERIMENTAL_RANGES` page(s) to sidebar nav

- Had to update full SQL diagram so the link to the `show_stmt_ranges`
  nonterminal would work properly

- Finally, replicated all changes into v2.1 folder

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/2930)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3f45481f6e36138f3570fc650881852e01c29544/

rmloveland commented:
@RaduBerinde and @lhirata :

Unfortunately this PR had to contain a lot of inconsequential administrative stuff around updating SQL diagrams and the 2.1 release.  Sorry about that.

To see the actual new doc, go here:

[SHOW EXPERIMENTAL_RANGES](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3f45481f6e36138f3570fc650881852e01c29544/stable/show-experimental-ranges.html)

RaduBerinde commented:
The doc looks great!

I would clarify a bit that when we say "Table" we really mean primary index; maybe just add `(Primary Index)` to the `Show Ranges For a Table` heading.

rmloveland commented:
> The doc looks great!

Thank you!

> maybe just add (Primary Index) to the Show Ranges For a Table heading.

Done!

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/8cae2ccb19160300a0270baa7269d8ccab18d405/

lhirata commented:
Review status: 0 of 18 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[_includes/sql/v2.0/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzNTMDPpZ6L582sws:-LANzNTMDPpZ6L582swt:b-8iq0g7) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.0/diagrams/show_ranges.html#L10)):*
> ```HTML
> <rect x="113" y="3" width="166" height="32"></rect>
> <rect x="111" y="1" width="166" height="32" class="nonterminal"></rect>
> <text class="nonterminal" x="121" y="21">EXPERIMENTAL_RANGES</text>
> ```

I think we usually unlink any variable names we change. However, for this example, you might be able to inline `EXPERIMENTAL_RANGES` and hide `TESTING_RANGES`

---

*[_includes/sql/v2.1/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzssN401pzUywRo6l:-LANzssN401pzUywRo6m:b-ya81ng) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.1/diagrams/show_ranges.html#L10)):*
> ```HTML
> <rect x="113" y="3" width="166" height="32"></rect>
> <rect x="111" y="1" width="166" height="32" class="nonterminal"></rect>
> <text class="nonterminal" x="121" y="21">EXPERIMENTAL_RANGES</text>
> ```

Same as above

---

*[v2.0/partitioning.md, line 232 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-6tt1n4lKfyuoY3R:-LAO-6tt1n4lKfyuoY3S:b-aigaia) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L232)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students_by_list;
> ```

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_list;

---

*[v2.0/partitioning.md, line 338 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-RxDDaR78qcctez4:-LAO-RxDDaR78qcctez5:bvzb68d) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L338)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students_by_range;
> ```

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_range;

---

*[v2.0/partitioning.md, line 480 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-XqaBQ0TDPVzewdD:-LAO-XqaBQ0TDPVzewdE:b-7czb0d) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L480)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students;
> ```

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students;

(this is probably something that should be explicit in our style guide. i'll note it)

---

*[v2.0/show-experimental-ranges.md, line 37 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-ul_02RVo8xsItU2:-LAO-ul_02RVo8xsItU3:b-1rbrxg) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L37)):*
> ```Markdown
> Parameter | Description
> ----------|------------
> [`table_name`](sql-grammar.html#table_name) | The name of the table you'd like to see range information for.
> ```

nit / suggestion: The name of the table you want range information about.

---

*[v2.0/show-experimental-ranges.md, line 38 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO0B6h9SWXHGQojmr6:-LAO0B6h9SWXHGQojmr7:b-7wuihs) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L38)):*
> ```Markdown
> ----------|------------
> [`table_name`](sql-grammar.html#table_name) | The name of the table you'd like to see range information for.
> [`table_name_with_index`](sql-grammar.html#table_name_with_index) | The name of the index you'd like to see range information for.
> ```

nit / suggestion: The name of the index you want range information about.

---

*[v2.0/show-experimental-ranges.md, line 52 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO1yK38upT_9TPEmI6:-LAO1yK38upT_9TPEmI7:bqmojkd) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L52)):*
> ```Markdown
>        zip_code INTEGER NOT NULL,
>        credit_score INTEGER NOT NULL,
>        PRIMARY KEY (zip_code)
> ```

(I might be misunderstanding so tell me if that's the case) A primary key must be unique, so designating the zip code as the primary key seems confusing since more than one user can have the same zip.

---

*[v2.0/show-experimental-ranges.md, line 124 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3qfS9ivboIsDHi0p:-LAO3qfS9ivboIsDHi0q:b4fedx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L124)):*
> ```Markdown
> - [`CREATE INDEX`](create-index.html)
> - [Indexes](indexes.html)
> + [Follow-the-workload](demo-follow-the-workload.html)
> ```

nit: Follow-the-Workload

---

*[v2.1/partitioning.md, line 232 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3GYNA7gsHOpXX80f:-LAO3GYNA7gsHOpXX80g:b-otiihc) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L232)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students_by_list;
> ```

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_list;

---

*[v2.1/partitioning.md, line 338 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3M4V7U2E0G9w8fm7:-LAO3M4V7U2E0G9w8fm8:ben2mwb) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L338)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students_by_range;
> ```

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_range;

---

*[v2.1/partitioning.md, line 480 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3Ol-1t57QwmTd3pT:-LAO3Ol-1t57QwmTd3pU:b-dw2ai1) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L480)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > show experimental_ranges from table students;
> ```

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students;

---

*[v2.1/show-experimental-ranges.md, line 37 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3Y-Z1DHhttr6_zVT:-LAO3Y-Z1DHhttr6_zVU:b-aeec6) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L37)):*
> ```Markdown
> Parameter | Description
> ----------|------------
> [`table_name`](sql-grammar.html#table_name) | The name of the table you'd like to see range information for.
> ```

see comment for 2.0

---

*[v2.1/show-experimental-ranges.md, line 38 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3ZfQ0Ppk0HCChxIC:-LAO3ZfQ0Ppk0HCChxID:b-aeec6) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L38)):*
> ```Markdown
> ----------|------------
> [`table_name`](sql-grammar.html#table_name) | The name of the table you'd like to see range information for.
> [`table_name_with_index`](sql-grammar.html#table_name_with_index) | The name of the index you'd like to see range information for.
> ```

see comment for 2.0

---

*[v2.1/show-experimental-ranges.md, line 124 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3x5x-QYVQGUNv3Gd:-LAO3x5x-QYVQGUNv3Ge:b4fedx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L124)):*
> ```Markdown
> - [`CREATE INDEX`](create-index.html)
> - [Indexes](indexes.html)
> + [Follow-the-workload](demo-follow-the-workload.html)
> ```

nit: Follow-the-Workload

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930)*
<!-- Sent from Reviewable.io -->

rmloveland commented:
Review status: 0 of 18 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v2.0/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzNTMDPpZ6L582sws:-LAOnKDXEJyGED8iLP7A:b-1b9ujk) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.0/diagrams/show_ranges.html#L10)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

I think we usually unlink any variable names we change. However, for this example, you might be able to inline `EXPERIMENTAL_RANGES` and hide `TESTING_RANGES`
</blockquote></details>

Sorry, I don't understand.  My questions:

1. The link from `EXPERIMENTAL_RANGES` appears to work as a pointer to the ranges statement block in the full grammar.  Are you saying we should make it not be a link to that?  (I can do this but would prefer not to since AFAICT it's linking to the right block in the grammar.  But there may be other considerations I don't know anything about, so please let me know.  I may also just be generally confused.)

2. What do you mean by "hide TESTING_RANGES"?  Sorry if this is a dumb question, I'm not seeing it on the page.

---

*[_includes/sql/v2.1/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzssN401pzUywRo6l:-LAP11SuA87cgCP1sYjZ:b75bs8c) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.1/diagrams/show_ranges.html#L10)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Responded above

---

*[v2.0/partitioning.md, line 232 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-6tt1n4lKfyuoY3R:-LAOoMgNCBr64C8o6xVR:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L232)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_list;
</blockquote></details>

Fixed

---

*[v2.0/partitioning.md, line 338 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-RxDDaR78qcctez4:-LAOoNfR5USJfDiLNrX8:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L338)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_range;
</blockquote></details>

Fixed

---

*[v2.0/partitioning.md, line 480 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-XqaBQ0TDPVzewdD:-LAOoPe578Ho9UXMgsoK:b5x8e48) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/partitioning.md#L480)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: CAPS for SHOW EXPERIMENTAL_RANGES FROM TABLE students;

(this is probably something that should be explicit in our style guide. i'll note it)
</blockquote></details>

Fixed.  In this instance I think they were already lower-cased so I left them that way, but in future I will update lowercase keywords to use caps when I come across them.

---

*[v2.0/show-experimental-ranges.md, line 37 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO-ul_02RVo8xsItU2:-LAOozPu3oJeQfiU50Cc:bt2vr0y) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L37)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit / suggestion: The name of the table you want range information about.
</blockquote></details>

Fixed!

---

*[v2.0/show-experimental-ranges.md, line 38 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO0B6h9SWXHGQojmr6:-LAOp-RJ9L3Ug0DSEFE_:bt2vr0y) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L38)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit / suggestion: The name of the index you want range information about.
</blockquote></details>

Fixed!

---

*[v2.0/show-experimental-ranges.md, line 52 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO1yK38upT_9TPEmI6:-LAOp1c90pEURmcShPYQ:b-bbu3qh) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L52)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

(I might be misunderstanding so tell me if that's the case) A primary key must be unique, so designating the zip code as the primary key seems confusing since more than one user can have the same zip.
</blockquote></details>

You are correct!  GREAT CATCH, THANK YOU.  I redesigned the table(s) to use small integers for the primary key and split on them instead.

---

*[v2.0/show-experimental-ranges.md, line 124 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3qfS9ivboIsDHi0p:-LAOqXa534SAH1ygUyMz:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.0/show-experimental-ranges.md#L124)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: Follow-the-Workload
</blockquote></details>

Fixed

---

*[v2.1/partitioning.md, line 232 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3GYNA7gsHOpXX80f:-LAOpBSw0vPt2LEGHU4z:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L232)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_list;
</blockquote></details>

Fixed

---

*[v2.1/partitioning.md, line 338 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3M4V7U2E0G9w8fm7:-LAOpCT76-wbKn_ViG-K:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L338)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students_by_range;
</blockquote></details>

Fixed

---

*[v2.1/partitioning.md, line 480 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3Ol-1t57QwmTd3pT:-LAOpE2VERffk68Yh6do:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/partitioning.md#L480)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: SHOW EXPERIMENTAL_RANGES FROM TABLE students;
</blockquote></details>

Fixed

---

*[v2.1/show-experimental-ranges.md, line 37 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3Y-Z1DHhttr6_zVT:-LAOpGNR9gX3uAyb00FU:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L37)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

see comment for 2.0
</blockquote></details>

Fixed

---

*[v2.1/show-experimental-ranges.md, line 38 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3ZfQ0Ppk0HCChxIC:-LAOpHFHCSWpoOUXuQMq:b-886gsx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L38)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

see comment for 2.0
</blockquote></details>

Fixed

---

*[v2.1/show-experimental-ranges.md, line 124 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LAO3x5x-QYVQGUNv3Gd:-LAOpn4RBud_ZycIDgcj:bt2vr0y) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/v2.1/show-experimental-ranges.md#L124)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

nit: Follow-the-Workload
</blockquote></details>

Fixed!

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/1e782f459cfac320fbb3f1ca28504343037b357e/

lhirata commented:
Review status: 0 of 18 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v2.0/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzNTMDPpZ6L582sws:-LAP2PutEoufWHtKE0SJ:bl5nwxx) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.0/diagrams/show_ranges.html#L10)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Sorry, I don't understand.  My questions:

1. The link from `EXPERIMENTAL_RANGES` appears to work as a pointer to the ranges statement block in the full grammar.  Are you saying we should make it not be a link to that?  (I can do this but would prefer not to since AFAICT it's linking to the right block in the grammar.  But there may be other considerations I don't know anything about, so please let me know.  I may also just be generally confused.)

2. What do you mean by "hide TESTING_RANGES"?  Sorry if this is a dumb question, I'm not seeing it on the page.
</blockquote></details>

Ahh, SQL Railroad diagram rejiggering fun. Let me see if I can explain myself here; otherwise, we can chat.

1. You're correct, the link works as expected and points to the correct statement block.  My thought is that we inline (re: "pull" or "surface") the EXPERIMENTAL_RANGES and TESTING_RANGES blocks into the show_ranges diagram so that it looks something like:

                 EXPERIMENTAL_RANGES
SHOW <                                              > FROM TABLE [...]
                     TESTING_RANGES

If you inline the two ranges, you would then need to hide TESTING_RANGES (it doesn't show up now currently - I meant it hypothetically). The end result would be SHOW EXPERIMENTAL_RANGES FROM TABLE [...]

To do this, you add inline and exclude lines in diagrams.go to the show_ranges_stmt object:

```inline: []string{"ranges_kw"},
exclude: []*regexp.Regexp{regexp.MustCompile("'TESTING_RANGES'")},
```


---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930)*
<!-- Sent from Reviewable.io -->

lhirata commented:
Review status: 0 of 18 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v2.0/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzNTMDPpZ6L582sws:-LAP5Zu2E2THRWfic3Kq:bqqmp3s) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.0/diagrams/show_ranges.html#L10)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Ahh, SQL Railroad diagram rejiggering fun. Let me see if I can explain myself here; otherwise, we can chat.

1. You're correct, the link works as expected and points to the correct statement block.  My thought is that we inline (re: "pull" or "surface") the EXPERIMENTAL_RANGES and TESTING_RANGES blocks into the show_ranges diagram so that it looks something like:

                 EXPERIMENTAL_RANGES
SHOW <                                              > FROM TABLE [...]
                     TESTING_RANGES

If you inline the two ranges, you would then need to hide TESTING_RANGES (it doesn't show up now currently - I meant it hypothetically). The end result would be SHOW EXPERIMENTAL_RANGES FROM TABLE [...]

To do this, you add inline and exclude lines in diagrams.go to the show_ranges_stmt object:

```inline: []string{"ranges_kw"},
exclude: []*regexp.Regexp{regexp.MustCompile("'TESTING_RANGES'")},
</blockquote></details>

blah, my comment didn't seem to render correctly:

```
                   EXPERIMENTAL_RANGES
SHOW <                                                  > FROM TABLE table [...]
                       TESTING_RANGES
```

To do this, you add inline and exclude lines in diagrams.go to the show_ranges_stmt object:

```

inline: []string{"ranges_kw"},
exclude: []*regexp.Regexp{regexp.MustCompile("'TESTING_RANGES'")},

```

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930)*
<!-- Sent from Reviewable.io -->

rmloveland commented:
Review status: 0 of 18 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v2.0/diagrams/show_ranges.html, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-LANzNTMDPpZ6L582sws:-LAPP7ZD9eemin8f4XsL:b-30wazm) ([raw file](https://github.com/cockroachdb/docs/blob/8cae2ccb19160300a0270baa7269d8ccab18d405/_includes/sql/v2.0/diagrams/show_ranges.html#L10)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

blah, my comment didn't seem to render correctly:

```
                   EXPERIMENTAL_RANGES
SHOW <                                                  > FROM TABLE table [...]
                       TESTING_RANGES
```

To do this, you add inline and exclude lines in diagrams.go to the show_ranges_stmt object:

```

inline: []string{"ranges_kw"},
exclude: []*regexp.Regexp{regexp.MustCompile("'TESTING_RANGES'")},

```

</blockquote></details>

Fixed.  Thank you for explaining this to me!

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930)*
<!-- Sent from Reviewable.io -->

lhirata commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 18 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2930#-:-LAPS9rP1Zi6KFqQ9USv:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jseldess commented:
@rmloveland, this is awesome. Just a few nits:

- Please add `SHOW EXPERIMENTAL_RANGES` to `sql-statements.md`. While you're at it, please add `EXPERIMENTAL AUDIT`.
- In 2.0, the statement is still called `SHOW TESTING_RANGES`, based on my testing (see below). So we need to update all of the affected 2.0 pages accordingly. 
- In the first 2.1 alpha, it seems to be called `SHOW TESTING_RANGES` still as well (see below). It might be that you've documented a change that hasn't made it into any official release yet. In this case, although not ideal, if you can verify that the change will be in the next 2.1 alpha (which is 4 weeks off...), it might be ok to leave as is. Thoughts?

**2.0**
```
~/Downloads/cockroach-v2.0.0.darwin-10.9-amd64$ ./cockroach version
Build Tag:    v2.0.0
Build Time:   2018/04/03 20:52:33
Distribution: CCL
Platform:     darwin amd64 (x86_64-apple-darwin13)
Go Version:   go1.10
C Compiler:   4.2.1 Compatible Clang 3.8.0 (tags/RELEASE_380/final)
Build SHA-1:  a6b498b7aff14234bcde23107b9e7fa14e6a34a8
Build Type:   release

~/Downloads/cockroach-v2.0.0.darwin-10.9-amd64$ ./cockroach sql --insecure
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
#
# Server version: CockroachDB CCL v2.0.0 (x86_64-apple-darwin13, built 2018/04/03 20:52:33, go1.10) (same version as client)
# Cluster ID: a215d968-0d4c-4c46-b508-a90fe16ee604
#
# Enter \? for a brief introduction.
#
warning: no current database set. Use SET database = <dbname> to change, CREATE DATABASE to make a new database.
root@:26257/> create database test;
CREATE DATABASE

Time: 5.237802ms

root@:26257/> create table test.t1 (a int primary key, b string);
CREATE TABLE

Time: 7.993414ms

root@:26257/> show testing_ranges from table test.t1;
+-----------+---------+----------+----------+--------------+
| Start Key | End Key | Range ID | Replicas | Lease Holder |
+-----------+---------+----------+----------+--------------+
| NULL      | NULL    |       22 | {1}      |            1 |
+-----------+---------+----------+----------+--------------+
(1 row)

Time: 6.06852ms

root@:26257/> show experimental_ranges from table test.t1;
invalid syntax: statement ignored: syntax error at or near "from"
DETAIL: source SQL:
show experimental_ranges from table test.t1;
                         ^
```

**2.1-alpha**

```
~/Downloads/cockroach-v2.1.0-alpha.20180416.darwin-10.9-amd64$ ./cockroach version
Build Tag:    v2.1.0-alpha.20180416
Build Time:   2018/04/13 13:30:12
Distribution: CCL
Platform:     darwin amd64 (x86_64-apple-darwin13)
Go Version:   go1.10
C Compiler:   4.2.1 Compatible Clang 3.8.0 (tags/RELEASE_380/final)
Build SHA-1:  f54870592a75b62bf8b04ab6eb1c74442b57164f
Build Type:   release

~/Downloads/cockroach-v2.1.0-alpha.20180416.darwin-10.9-amd64$ ./cockroach sql --insecure
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
#
# Server version: CockroachDB CCL v2.1.0-alpha.20180416 (x86_64-apple-darwin13, built 2018/04/13 13:30:12, go1.10) (same version as client)
# Cluster ID: c47bfb73-b9e7-43c5-b135-249eb4aced67
#
# Enter \? for a brief introduction.
#
warning: no current database set. Use SET database = <dbname> to change, CREATE DATABASE to make a new database.
root@:26257/> create database test;
CREATE DATABASE

Time: 4.492528ms

root@:26257/> create table test.t1 (a int primary key, b string);
CREATE TABLE

Time: 3.802014ms

root@:26257/> show testing_ranges from table test.t1;
+-----------+---------+----------+----------+--------------+
| Start Key | End Key | Range ID | Replicas | Lease Holder |
+-----------+---------+----------+----------+--------------+
| NULL      | NULL    |       22 | {1}      |            1 |
+-----------+---------+----------+----------+--------------+
(1 row)

Time: 3.882265ms

root@:26257/> show experimental_ranges from table test.t1;
invalid syntax: statement ignored: syntax error at or near "from"
DETAIL: source SQL:
show experimental_ranges from table test.t1;
                         ^
```

rmloveland commented:
Thanks for the review @jseldess! Sorry for the errors.  Opened a new issue at https://github.com/cockroachdb/docs/issues/2954 to address.
